File: 1. Escuela Desarrollo Web/25-curso-POO/CursoPOOUber/JS/Car.js
class Car {

    constructor(license, driver) {
        this.id;
        this.license = license;
        this.driver = driver;
        this.passenger;
    }

    printDataCar() {
        console.log(this.driver, '    <- this.driver');
        console.log(this.driver.name, '    <- this.driver.name');
        console.log(this.driver.document, '    <- this.driver.document');
    }
}


// ! Old Way

// function Car(license, driver) {
//     this.id;
//     this.license = license;
//     this.driver = driver;
//     this.passenger;
// }

// Car.prototype.printDataCar = function() {
//     console.log(this.driver);
//     console.log(`this.driver.name: ${this.driver.name}`);
//     console.log(`this.driver.document: ${this.driver.document}`);
// }

File: 1. Escuela Desarrollo Web/25-curso-POO/CursoPOOUber/Java/Car.java
public class Car {
    Integer id;
    String license;
    Account driver;
    Integer passengers;

    public Car(String license, Account driver) {
        this.license = license;
        this.driver = driver;
    }

    void printDataCar() {
        System.out.println(
        "id: " + id + " " + 
        "Licence: " + license + " " + 
        "Driver: " + driver.name + " " + 
        "Passengers: " + passengers
        );
    }
}
